### DISCUSSION TITLE
Detailed parameter description is missing in Swagger UI

### DISCUSSION THREAD

**Discussion by megalaren** [fastapi/fastapi#10254] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/items/")
async def read_items(q: Annotated[list[str] | None, Query()] = None):
    query_items = {"q": q}
    return query_items
```

### Description

I'm repeating the code from here:
https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#query-parameter-list-multiple-values

What do I expect to see:
![1](https://github.com/tiangolo/fastapi/assets/67578764/8390ce0c-67fd-4117-be89-575728dc27bf)

What I see in reality::
![2](https://github.com/tiangolo/fastapi/assets/67578764/a8d0bea7-6efb-4aa4-ab18-3f88fd3c5017)

### Operating System

Linux

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.11

**Comment by hanouchen** [comment]
I am facing the same issue, removing the `| None` in the annotation makes it work for me but looks like some sort of bug?
```
@app.get("/items/")
async def read_items(q: Annotated[list[str], Query()] = None):
    query_items = {"q": q}
    return query_items
```
Also, this only seem to happen with pydantic V2.

**Comment by White-Mask** [comment]
Hello @megalaren, I tried to simulate the error on an Ubuntu system, and indeed, when using 'Annotated', the expected result is not achieved.
Nevertheless, you can opt for the option without 'Annotated'.
```python
@app.get(
    "/items/",
    summary="Get a list of items",
    description="This route retrieves a list of items.",
)
async def read_items(
    q: list[str] = Query(None, description="Description of the 'q' parameter")
):
    query_items = {"q": q}
    return query_items
```
You can also use the option without 'Annotated,' and as @hanouchen  suggested, removing 'None' from 'Annotated' also yields the expected result.
```python
@app.get("/items/")
async def read_items(q: Annotated[list[str], Query()] = None):
    query_items = {"q": q}
    return query_items
```
Lastly, another option to achieve the same result is to use 'or' instead of '|'.
```python
@app.get("/items/")
async def read_items(q: Annotated[list[str] or None, Query()] = None):
    query_items = {"q": q}
    return query_items
```
Regards.

**Reply by megalaren** [reply]
Thanks for the possible solutions! But I would like them to fix this in future versions.

**Comment by jaredb1011** [comment]
I just ran into the same bug while working through the tutorial and can confirm this is still an issue as of 11/8/23

**Comment by megalaren** [comment, marked as answer]
Just want to say that this issue has been fixed in version 0.111.1 by upgrading the Swagger-UI version
